src/main/java/calculator/Application.java

```diff
@@ -1,7 +1,157 @@
 package calculator;
 
+import camp.nextstep.edu.missionutils.Console;
+
+/**
+ * 이 클래스는 문자열 덧셈 계산기를 나타낸다.
+ * 커스텀 구분자가 있는지 확인하는 메서드, 구분자 배열을 반환하는 메서드,
+ * 커스텀 구분자 입력 부분을 제거하는 메서드, 구분자를 콤마로 대체하는 메서드,
+ * 문자열을 구분자를 기준으로 나누는 메서드, 입력값의 유효성을 검증하는 메서드,
+ * 배열의 합을 반환하는 메서드로 구성된다.
+ */
 public class Application {
+    /** 입력 문자열에 커스텀 구분자가 있는지 확인하여 bool 값을 반환한다.
+     * 여기서 커스텀 구분자란 입력 문자열 앞 부분의 "//"와 "\n" 사이에 위치하는 문자를 말한다.
+     *
+     * @param input 입력 문자열
+     * @return 커스텀 구분자 존재 여부
+     */
+    public static boolean hasCustomSeparator(String input) {
+//        if (input.startsWith("//\\n")) { // empty separator 비허용 시
+//            return false;
+//        }
+        return input.startsWith("//") && input.contains("\\n");
+    }
+
+    /**
+     * 입력 문자열에서 ","를 제외한 구분자를 배열로 반환한다.
+     * 즉, 커스텀 구분자가 없다면 ":"만, 있다면 ":"와 해당 구분자를 포함한 배열을 반환한다.
+     *
+     * @param input 입력 문자열
+     * @return 구분자의 배열
+     */
+    public static String[] getSeparators(String input) {
+        String[] separators = new String[]
+                {":"};
+        if (hasCustomSeparator(input)) {
+            separators =
+                    new String[]{":",
+                            input.substring(2,
+                                    input.indexOf("\\n"))};
+        }
+        return separators;
```

Comment:
```suggestion
if(!hasCustomSeparator(input)){ return separators; }
return new String[]{":",  input.substring(2,input.indexOf("\\n"))};
```

이런 식의 구조를 사용하면 들여쓰기가 적어서 좀 더 가독성이 오르지않을까? 하고 생각됩니다.

추가로 정규표현식을 활용하여 substring 없이 이를 처리할 수 있습니다.

Reply:
조건 반전을 통해 들여쓰기를 좀 줄여볼 수 있군요. 감사합니다!
어차피 메서드 이름에서 어느 정도 의미 전달이 될테니, separator 변수를 아예 없애는 것도 고려해보려고 합니다.

정규식의 존재는 알고 있으나 basic하게 짜본다고 사용하지 않았는데,
정규식 공부도 할 겸, 정규식을 쓰는 코드도 작성해보면 좋겠네요.